Blocks/HTTP/impl/posix_server.h

```diff
@@ -262,7 +262,8 @@ class HTTPServerPOSIX final {
           if (!content_type.empty()) {
             // `url_dirname` has no trailing slash.
             const std::string url_dirname =
-                options.url_base + current::strings::Join(item_info.path_components_cref, '/');
+                options.url_base + (options.url_base == "/" || item_info.path_components_cref.empty() ? "" : "/") +
```

Comment:
Extra parentheses to stress the `||` is to the left of `?` would help here. LGTM! Feel free to merge yourself, making you an admin now.